src/components/Exchange/PositionSeller.js

```diff
@@ -270,19 +340,52 @@ export default function PositionSeller(props) {
       }
     }
 
+    receiveToken = (allowReceiveTokenChange && swapToken) 
+      ? swapToken
+      : collateralToken;
+
     receiveAmount = receiveAmount.add(collateralDelta);
+    receiveAmount = receiveAmount.mul(bigNumberify(1000000));
+    convertedReceiveAmount = getTokenAmount(receiveAmount, receiveToken.address, false, infoTokens);
+
+    if (allowReceiveTokenChange && swapToken) {
+      const swapTokenInfo = getTokenInfo(infoTokens, swapToken.address);
+
+      notEnoughReceiveTokenLiquidity = swapTokenInfo.availableAmount.lt(convertedReceiveAmount);
+
+      const { feeBasisPoints } = getNextToAmount(
+        chainId,
+        fromAmount,
+        collateralToken.address,
+        receiveToken.address,
+        infoTokens,
+        undefined,
+        undefined,
+        usdgSupply,
+        totalTokenWeights,
+        true
+      );
+
+      if (feeBasisPoints) {
+        const baseFromAmountUsd = fromAmount.mul(BASIS_POINTS_DIVISOR).div(position.leverage)
+        swapFeeUsd = baseFromAmountUsd.mul(feeBasisPoints).div(BASIS_POINTS_DIVISOR)
+        swapFee = getTokenAmount(swapFeeUsd, collateralToken.address, false, infoTokens)
+      }
+    }
+
+    totalFees = totalFees
+      .add(positionFee || bigNumberify(0))
+      .add(fundingFee || bigNumberify(0))
+      .add(swapFeeUsd || bigNumberify(0))
```

Comment:
positionFee and fundingFee is denominated in USD
swapFee is denominated in token
swapFeeUsd is denominated in USD

maybe to avoid confusion better to name `swapFee` for USD denominated and `swapFeeToken` (swapFeeInToken?), what do you think?